assets/app/scripts/controllers/pod.js

```diff
@@ -88,6 +88,26 @@ angular.module('openshiftConsole')
           }
         );
 
+
+        // outside of the above watch to avoid repeatedly generating new watches.
+        $scope.$watch('logOptions.container', function(newVal) {
+          if(!newVal) {
+            return;
+          }
+         var containerStatus = _.find(
+                                  $scope.pod.status.containerStatuses,
+                                  function(container) {
+                                  return container.name === newVal;
+                                });
+
+          var state = _.get(containerStatus, 'state');
+          var status = _.first(_.keys(state));
+          // need to end with these three things defined to pass to the directive.
+          $scope.status = _.includes(['running', 'waiting', 'terminated'], status) ? _.capitalize(status) : '';
+          $scope.startTime = state[status].startedAt;
+          $scope.endTime = state[status].finishedAt;
```

Comment:
You want to specifically catch the situation where current status is waiting. Then you should use the start and end time from lastState -- if defined -- since that's the start and end times of the log we show.